src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/EditIssuePage.kt

```diff
@@ -49,4 +53,52 @@ class EditIssuePage(
             webElement.sendKeys(text)
         }
     }
+
+}
+
+object PageRteUtil{
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    fun clearAndTypeIfPresent(driver: WebDriver, id :String, text: String) {
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            driver.wait(timeout = Duration.ofSeconds(2), condition = elementToBeClickable(webElement))
+            webElement.clear()
+            webElement.sendKeys(text)
+        }
+    }
+
+    fun clearAndTypeIfPresentRte(driver: WebDriver,id : String, text: String) {
+        //check richeditor-cover exists in class attribute
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            val attrClass = webElement.getAttribute("class")
+            if(attrClass != null && attrClass.contains(other = "richeditor-cover", ignoreCase = true)){
+                logger.debug("richeditor detected, will try Rich Text Editer...")
+                val iframeXpath = "//div[textarea[@id='$id']]//iframe"
+                driver.wait(timeout = Duration.ofSeconds(10), condition = frameToBeAvailableAndSwitchToIt(By.xpath(iframeXpath)))
+
+                val bodyElement = driver.findElement(By.id("tinymce"))
+                //data-projectkey={project}
+                driver.wait(timeout = Duration.ofSeconds(10), condition = ExpectedConditions.attributeToBeNotEmpty(bodyElement, "data-projectkey"))
+                bodyElement.clear()
+                bodyElement.sendKeys(text)
+
+                driver.switchTo().parentFrame()
+            }
+            else{
+                //try plain text editing
+                logger.debug("Try use Raw Text Editer...")
+                clearAndTypeIfPresent(driver, id, text)
+            }
+        }
+    }
+
+    private fun logElement(bodyElement: WebElement, driver: WebDriver) {
```

Comment:
We should clean up the temporary debugging constructs

Reply:
yes